reverse_proxy.go

```diff
@@ -253,6 +250,10 @@ func TykNewSingleHostReverseProxy(target *url.URL, spec *APISpec) *ReverseProxy
 			// this line, we would get the net/http default.
 			req.Header.Set("User-Agent", defaultUserAgent)
 		}
+
+		if spec.Proxy.SkipTargetPathEscaping {
+			req.URL.Opaque = req.URL.RawPath // if we set opaque here it will force URL.RequestURI to skip escaping
```

Comment:
I think this change is not gonna work as `req.URL.RawPath` contains at this point full initial request URI - listen path + the rest of the path

Reply:
another note - `req.URL.RawPath` is going to have query string (if any) `?abc=xyz` and `RequestURI()` is adding query string here to result of escape-call again: https://golang.org/src/net/url/url.go#L994
if I understand right - we will have get passed to upstream result of `req.URL.Opaque + "?" + u.RawQuery` with `?abc=xyz` doubled.

Reply:
From what I understand RawPath do not include query string

Reply:
ok, lets scratch my comment about query string - it is not included to `RawPath`
I still see issue with setting `req.URL.Opaque = req.URL.RawPath` as it contains at this point initial request path (incoming to gateway) and leads to sending it to upstream as is.

the changes https://github.com/TykTechnologies/tyk/commit/a7a2a04107400cc7f1b60fce81b2ae945d11bd04 @lonelycode submitted should address this issue as we remove listen path, and replace in requests URL with object (result of parsing striped URI)

Reply:
Pls update your PR to use martin changes if you thin they will work.

Reply:
👍